sds_data_manager/stacks/sqs_stack.py

```diff
@@ -0,0 +1,75 @@
+"""Stack for creating instrument SQS queues and attaching batch_starter as a target."""
+
+from aws_cdk import Duration, Stack, aws_sqs
+from aws_cdk import aws_events as events
+from aws_cdk import aws_events_targets as targets
+from constructs import Construct
+
+
+class SqsStack(Stack):
+    """Stack to create instrument/level SQS queues and attach them to EventBridge."""
+
+    def __init__(
+        self,
+        scope: Construct,
+        construct_id: str,
+        instrument_names: list[str],
+        **kwargs,
+    ):
+        """Create a stack to create a SQS queue and Eventbridge rule for an instrument.
+
+        Parameters
+        ----------
+        scope : Construct
+            Parent construct.
+        construct_id : str
+            A unique string identifier for this construct.
+        instrument_names : list[str]
+            A list of all instrument names
+        kwargs : dict
+            Keyword arguments
+        """
+        super().__init__(scope, construct_id, **kwargs)
+        # This needs to be a FIFO queue to enforce ordering
+        self.instrument_queue = aws_sqs.Queue(
+            self,
+            "FileArrivalQueue",
+            queue_name="file_arrival_queue.fifo",
+            fifo=True,
+            encryption=aws_sqs.QueueEncryption.UNENCRYPTED,
+            # This timeout determines how long the queue waits for processing. It must
+            # be longer than the timeout of the lambda.
+            visibility_timeout=Duration.seconds(300),
+            # This is required. It removes messages with identical content. Since
+            # the event includes a filename each event should be totally unique.
+            content_based_deduplication=True,
+        )
+
+        for instrument_name in instrument_names:
```

Comment:
It will slow processing down. I would be surprised if it ended up causing issues, but we can also change settings to reduce that issue (for example by batching processing together to eliminate lambda startup times) or add additional queues

Reply:
Ultimately this is not going to be a bottleneck for processing because the actual processing steps still run in parallel (with a few seconds delayed starting) and they will take much longer than triggering jobs will.